File: app/lib/models/face.dart
class FaceCluster {
  final String id;
  final String? name;
  final int photoCount;
  final String? coverPhotoId;
  final String? coverThumbUrl;
  final String createdAt;

  FaceCluster({
    required this.id,
    this.name,
    required this.photoCount,
    this.coverPhotoId,
    this.coverThumbUrl,
    this.createdAt = '',
  });

  factory FaceCluster.fromJson(Map<String, dynamic> json) {
    return FaceCluster(
      id: json['id'],
      name: json['name'],
      photoCount: json['photo_count'] ?? 0,
      coverPhotoId: json['cover_photo_id'],
      coverThumbUrl: json['cover_thumb_url'],
      createdAt: json['created_at'] ?? '',
    );
  }
}

class AIStatus {
  final bool aiAvailable;
  final int queueSize;
  final int totalFaces;
  final int namedFaces;
  final int totalPhotoFaces;

  AIStatus({
    required this.aiAvailable,
    required this.queueSize,
    required this.totalFaces,
    required this.namedFaces,
    required this.totalPhotoFaces,
  });

  factory AIStatus.fromJson(Map<String, dynamic> json) {
    return AIStatus(
      aiAvailable: json['ai_available'] ?? false,
      queueSize: json['queue_size'] ?? 0,
      totalFaces: json['total_faces'] ?? 0,
      namedFaces: json['named_faces'] ?? 0,
      totalPhotoFaces: json['total_photo_faces'] ?? 0,
    );
  }
}

File: server/schemas/face.py
"""Face API schemas."""

from pydantic import BaseModel


class FaceResponse(BaseModel):
    id: str
    name: str | None
    photo_count: int
    cover_photo_id: str | None
    cover_thumb_url: str | None = None
    created_at: str = ""


class FaceListResponse(BaseModel):
    faces: list[FaceResponse]
    total: int


class FaceTagRequest(BaseModel):
    name: str


class FaceMergeRequest(BaseModel):
    source_face_ids: list[str]


class FaceMergeResponse(BaseModel):
    id: str
    name: str | None
    photo_count: int
    merged: int


class FaceReclusterResponse(BaseModel):
    clusters: int
    merged: int


class AIStatusResponse(BaseModel):
    ai_available: bool
    queue_size: int
    total_faces: int
    named_faces: int
    total_photo_faces: int
